### PR TITLE
feat(remotecontrol): multi monitor support

### PR DESCRIPTION
**Pull request by hristoterezov**
Depends on https://github.com/jitsi/jitsi-meet-electron/pull/13

### CODE REVIEW

File: conference.js [hunk 1]
```diff
@@ -2148,5 +2148,15 @@ export default {
...
+    },
+
+    /**
+     * Returns the desktop sharing source id or null if the desktop sharing is
+     * not active at the moment.
+     *
+     * @returns {string} - The source id.
+     */
+    getDesktopSharingSourceId() {
+        return localVideo.sorceId;
```
Comment (virtuacoplenny) [inline conference.js]: I think there's a typo here with "sorceId"

File: conference.js [hunk 2]
```diff
@@ -2148,5 +2148,15 @@ export default {
             room.setDisplayName(formattedNickname);
             APP.UI.changeDisplayName(this.getMyUserId(), formattedNickname);
         }
+    },
+
+    /**
+     * Returns the desktop sharing source id or null if the desktop sharing is
+     * not active at the moment.
+     *
+     * @returns {string} - The source id.
```
Comment (virtuacoplenny) [inline conference.js]: Should probably update the doc to signify null can be returned (or would it be undefined that's returned?).

File: modules/remotecontrol/Receiver.js [hunk 3]
```diff
@@ -284,6 +284,19 @@ export default class Receiver extends RemoteControlParticipant {
...
+        transport.sendEvent({
+            name: REMOTE_CONTROL_EVENT_NAME,
+            type: EVENT_TYPES.start,
+            id: APP.conference.getSourceId()
+        });
+    }
+
+    /**
      * Handles remote control events from the external app. Currently only
      * events with type = EVENT_TYPES.supported or EVENT_TYPES.permissions.
```
Comment (virtuacoplenny) [inline modules/remotecontrol/Receiver.js]: Probably need to update this comment too because you changed supported event types.

**Comment by hristoterezov** [comment]
@virtuacoplenny I've just addressed all comments.

File: conference.js [hunk 4]
```diff
@@ -2117,5 +2117,16 @@ export default {
             room.setDisplayName(formattedNickname);
             APP.UI.changeDisplayName(this.getMyUserId(), formattedNickname);
         }
+    },
+
+    /**
+     * Returns the desktop sharing source id or null if the desktop sharing is
```
Comment (virtuacoplenny) [inline conference.js]: sharing source id or undefined*

**Comment by virtuacoplenny** [comment]
Are there jitsi-meet-electron changes that will be needed to make this work? (I think you answered this for me in person but I've forgotten the answer.) If so, should they be merged at the same time?

File: modules/remotecontrol/Receiver.js [hunk 5]
```diff
@@ -268,6 +267,7 @@ export default class Receiver extends RemoteControlParticipant {
                         type: EVENT_TYPES.permissions,
                         action: PERMISSIONS_ACTIONS.grant
                     });
+                    this._sendStartAPIEvent();
```
Comment (virtuacoplenny) [inline modules/remotecontrol/Receiver.js]: If the grant action and the api start event are always coupled, maybe they should be broken out into a function to encapsulate the behavior.

Comment (hristoterezov) [inline modules/remotecontrol/Receiver.js]: I refactored this a little.

**Comment by hristoterezov** [comment]
@virtuacoplenny I'm working on the electron part of it. This PR and the electron one should be merged together. I added 'WIP' in the description.

File: modules/remotecontrol/Receiver.js [hunk 6]
```diff
@@ -255,50 +211,63 @@ export default class Receiver extends RemoteControlParticipant {
...
-                            type: EVENT_TYPES.permissions,
-                            action: PERMISSIONS_ACTIONS.error
-                        });
-                    }
+            promise = APP.conference.toggleScreenSharing()
+                .then(() => this._sendStartRequest());
+        }
+
+        promise
+            .than(() =>
```
Comment (virtuacoplenny) [inline modules/remotecontrol/Receiver.js]: I just noticed this typo. This should be .then.

**Comment by virtuacoplenny** [comment]
Also, not part of these changes but related, the unbinding of mouse event listeners in controller is not working. On stop of remote control, pause() is called which removes the listeners but it no-ops if  _controlledParticipant is falsy. _controlledParticipant is set to null right before pause is called.

File: modules/remotecontrol/Receiver.js (hunk 6 above)
Comment (hristoterezov) [inline modules/remotecontrol/Receiver.js]: Oh! I've forgotten to push this! Thanks for pointing it!

Comment (hristoterezov) [inline modules/remotecontrol/Receiver.js]: Fixed!

**Comment by hristoterezov** [comment]
@virtuacoplenny 
> Also, not part of these changes but related, the unbinding of mouse event listeners in controller is not working. On stop of remote control, pause() is called which removes the listeners but it no-ops if _controlledParticipant is falsy. _controlledParticipant is set to null right before pause is called.

Fixed!

**Pull request merged**